ethcore/src/client/client.rs

```diff
@@ -1245,7 +1245,9 @@ impl BlockChainClient for Client {
 	}
 
 	fn chain_info(&self) -> BlockChainInfo {
-		self.chain.read().chain_info()
+		let mut chain_info = self.chain.read().chain_info();
+		chain_info.pending_total_difficulty = chain_info.total_difficulty + self.block_queue.total_difficulty();
```

Comment:
This is used to make a decision whether to try and sync to a peer for a full sync. When the queue is full we don't care much if this is precise. When it is empty or nearly empty and we are far behind we don't care either. For the up-to date sync to unknown hashes this is not used because there's no way to get other peer's total difficulty. We treat it as unknown and try and download these new blocks anyway.